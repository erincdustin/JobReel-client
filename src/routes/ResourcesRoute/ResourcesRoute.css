.resources-page {
  display: grid;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  grid-gap: 1em;
}

.resourcesBack {
  position: relative;
  grid-column-start: 1;
  grid-column-end: 9;
}

.resourcesBack > img {
  height: 250px;
  width: 250px;
  position: absolute;
  top: -170px;
  left: 10%;
  z-index: -1;
}

@media (max-width: 960px) {
  .SideNav{
    display: none;
  }
}

@media (min-width: 960px) {
  .TopNav{
    display: none;
  }
}