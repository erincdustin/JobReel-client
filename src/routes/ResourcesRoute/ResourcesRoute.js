import React from 'react';
import SideNav from '../../components/SideNav/SideNav';

class ResourcesRoute extends React.Component {
  static defaultProps = {
    location: {},
    history: {
      push: () => { },
    },
  }

  state = { error: null };

  render() {
    return (
      <div className='resources-page'>
        <div className='title'>
          <h2>Resources</h2>
        </div>
        <SideNav />
        {/* <ResourcesForm/> */}
      </div>
    )
  }
}

export default ResourcesRoute;