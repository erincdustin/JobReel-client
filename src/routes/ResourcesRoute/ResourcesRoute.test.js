import React from 'react'
import ReactDOM from 'react-dom'
import ResourcesRoute from './ResourcesRoute'
import {BrowserRouter} from 'react-router-dom'
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import PropTypes from 'prop-types';

class ContextProvider extends React.Component {
  static childContextTypes = {
    myContext: PropTypes.object,
  }

  getChildContext = () => ({
    myContext: {
      manualResourceAdd: true
    }
  })

  render() {
    return this.props.children;
  }
}

describe('ResourcesRoute smoke test', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<BrowserRouter><ResourcesRoute /></BrowserRouter>, div)
    ReactDOM.unmountComponentAtNode(div)
  })
})

describe('ResourcesRoute snapshot test', () => {
  it('renders the UI as expected', () => {
    const wrapper = shallow(
      <ResourcesRoute />
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  
  it('renders the resources form when manually adding resources', () => {
    const wrapper = shallow(
      <ContextProvider>
        <ResourcesRoute />
      </ContextProvider>
    )
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})