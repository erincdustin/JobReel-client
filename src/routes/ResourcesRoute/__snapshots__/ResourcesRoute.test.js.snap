// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResourcesRoute snapshot test renders the UI as expected 1`] = `
<div
  className="resources-page"
>
  <div
    className="title"
  >
    <h2>
      Resources
    </h2>
  </div>
  <div
    className="resourcesBack"
  >
    <img
      alt="resources-background"
      src="resourcesBack.svg"
    />
  </div>
  <MediaQueryContextConsumer
    minDeviceWidth={961}
  >
    <SideNav />
  </MediaQueryContextConsumer>
  <MediaQueryContextConsumer
    maxDeviceWidth={960}
  >
    <TopNav />
  </MediaQueryContextConsumer>
  <ResourcesForm />
</div>
`;

exports[`ResourcesRoute snapshot test renders the resources form when manually adding resources 1`] = `<ResourcesRoute />`;
